tests/unit/core/compile/sqlglot/test_compiler.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import google.cloud.bigquery as bigquery
+import pandas as pd
+import pytest
+
+import bigframes.core
+import bigframes.core.nodes as nodes
+import bigframes.pandas as bpd
+
+
+def test_compile_local(
+    inline_pd_df: pd.DataFrame, sql_compiler_session: bigframes.Session
+):
+    bf_df = bpd.DataFrame(inline_pd_df, session=sql_compiler_session)
+    expected_sql = "SELECT * FROM UNNEST(ARRAY<STRUCT<`level_0` INT64, `column_0` INT64, `column_1` INT64, `column_2` BOOLEAN, `column_3` STRING>>[(0, 1, -10, TRUE, 'b'), (1, 2, 20, CAST(NULL AS BOOLEAN), 'aa'), (2, 3, 30, FALSE, 'ccc')]) AS `table_alias`"
+    assert bf_df.sql == expected_sql
```

Comment:
Neat!

I'm looking forward to seeing more of these sorts of "golden SQL" tests in future.